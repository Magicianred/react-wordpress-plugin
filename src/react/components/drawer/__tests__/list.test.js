import React from 'react'
import { shallow } from 'enzyme'
import StarIcon from '@material-ui/icons/Star'
import ListItem from '@material-ui/core/ListItem'

import DrawerList from '../list'

describe('Components/DrawerList', () => {
  const props = {
    list: [
      {
        name: 'Settings',
        icon: StarIcon,
      },
    ],
    onClick: jest.fn(),
  }

  beforeEach(() => {
    props.onClick.mockReset()
  })

  it('renders component', () => {
    expect(shallow(<DrawerList {...props} />)).toMatchSnapshot()
  })

  it("it's listItems invoke onClick", () => {
    const wrapper = shallow(<DrawerList {...props} />)
    const element = wrapper.find(ListItem)
    element.simulate('click')
    expect(props.onClick).toHaveBeenCalledTimes(1)
    expect(props.onClick).toHaveBeenCalledWith(0)
  })
})
