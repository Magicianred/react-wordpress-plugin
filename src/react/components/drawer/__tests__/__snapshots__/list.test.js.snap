// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components/DrawerList renders component 1`] = `
<React.Fragment>
  <WithStyles(ListItem)
    button={true}
    key="Settings"
    onClick={[Function]}
  >
    <WithStyles(ListItemIcon)>
      <pure(Star) />
    </WithStyles(ListItemIcon)>
    <WithStyles(ListItemText)
      primary="Settings"
    />
  </WithStyles(ListItem)>
</React.Fragment>
`;
